dotnet/src/Connectors/Connectors.Amazon.UnitTests/Settings/BedrockChatCompletionModelExecutionSettingsTests.cs

```diff
@@ -0,0 +1,432 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Amazon.Runtime.Endpoints;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Moq;
+using Xunit;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.UnitTests;
+
+/// <summary>
+/// Unit tests for prompt execution settings confirgurations for different Bedrock Models.
+/// </summary>
+public class BedrockChatCompletionModelExecutionSettingsTests
+{
+    /// <summary>
+    /// Checks that an invalid prompt execution settings will throw an exception.
+    /// </summary>
+    [Fact]
+    public async Task ShouldThrowExceptionForInvalidPromptExecutionSettingsAsync()
+    {
+        // Arrange
+        string modelId = "amazon.titan-text-lite-v1";
+        var mockBedrockApi = new Mock<IAmazonBedrockRuntime>();
+        mockBedrockApi.Setup(m => m.DetermineServiceOperationEndpoint(It.IsAny<ConverseRequest>()))
+            .Returns(new Endpoint("https://bedrock-runtime.us-east-1.amazonaws.com")
+            {
+                URL = "https://bedrock-runtime.us-east-1.amazonaws.com"
+            });
+        var invalidSettings = new AmazonTitanExecutionSettings()
+        {
+            Temperature = -1.0f,
+            TopP = -0.5f,
+            MaxTokenCount = -100
+        };
+        var kernel = Kernel.CreateBuilder().AddBedrockChatCompletionService(modelId, mockBedrockApi.Object).Build();
+        var service = kernel.GetRequiredService<IChatCompletionService>();
+        var chatHistory = CreateSampleChatHistory();
+
+        // Act & Assert
+        await Assert.ThrowsAsync<ArgumentException>(() =>
+            service.GetChatMessageContentsAsync(chatHistory, invalidSettings)).ConfigureAwait(true);
+    }
+
+    /// <summary>
+    /// Checks that the prompt execution settings are correctly registered for the chat completion call.
+    /// </summary>
+    [Fact]
+    public async Task TitanGetChatMessageContentsAsyncShouldReturnChatMessageWithPromptExecutionSettingsAsync()
+    {
+        // Arrange
+        string modelId = "amazon.titan-text-lite-v1";
+        var mockBedrockApi = new Mock<IAmazonBedrockRuntime>();
+        var executionSettings = new AmazonTitanExecutionSettings()
+        {
+            Temperature = 0.3f,
+            TopP = 0.8f,
+            MaxTokenCount = 510,
+            ModelId = modelId,
+            ExtensionData = new Dictionary<string, object>()
+            {
+                { "temperature", 0.3f },
+                { "topP", 0.8f },
+                { "maxTokenCount", 510 }
+            }
```

Comment:
> [!IMPORTANT]
> Apply for all.

Suggest in the internal impl of the `ExecutionSettings` type, adding and retrieving automatically from the `ExtensionData` this way the caller don't need to provide twice this value.

When the caller provides the extension data, it should override the `Property` (because the property actually should get from the `ExtensionData` behind the scenes.

Reply:
Apologies this test was unclear. Internal implementation of ExecutionSettings already does this- fixing unit tests to reflect this